extensions/wikia/Staging/Staging.setup.php

```diff
@@ -19,3 +19,15 @@
 
 $wgHooks['MakeGlobalVariablesScript'][] = 'StagingHooks::onMakeGlobalVariablesScript';
 $wgHooks['BeforePageRedirect'][] = 'StagingHooks::onBeforePageRedirect';
+
+/**
+ * Vary memcache by environment
+ *
+ * We append wfWikiID() here as wfMemcKey() uses
+ * $wgCachePrefix or wfWikiID() if the first one is not set
+ *
+ * @author macbre
+ * @see PLATFORM-664
+ */
+$wgCachePrefix = gethostname() . '-' . wfWikiID(); // e.g. staging-s3-muppet / sandbox-qa02-glee / ...
```

Comment:
Let's use a different separator to make this environment specific prefix stand out from the rest of the key.